modules/tls/src/main/java/org/mule/runtime/module/tls/internal/DefaultTlsContextFactory.java

```diff
@@ -213,12 +218,24 @@ public SSLContext createSslContext() throws KeyManagementException, NoSuchAlgori
   }
 
   @Override
+  public SSLSocketFactory getSocketFactory() throws KeyManagementException, NoSuchAlgorithmException {
+    return new RestrictedSSLSocketFactory(createSslContext(), getEnabledCipherSuites(), getEnabledProtocols());
+  }
+
+  @Override
+  public SSLServerSocketFactory getServerSocketFactory() throws KeyManagementException, NoSuchAlgorithmException {
+    return new RestrictedSSLServerSocketFactory(createSslContext(), getEnabledCipherSuites(), getEnabledProtocols());
+  }
+
+  @Override
   public String[] getEnabledCipherSuites() {
-    if (isUseDefaults(enabledCipherSuites)) {
-      return tlsConfiguration.getEnabledCipherSuites();
+    String[] enabledCipherSuites;
+    if (isUseDefaults(this.enabledCipherSuites)) {
```

Comment:
Question: When FIPS is enabled in the runtime, does it means that not all cipher suites can be used?. If that's the case, and the user configures a cipher suite not supported by FIPS, are we failing in that case?